src/lib/mo-pdf-viewer.component.ts

```diff
@@ -1,102 +1,143 @@
+  NgxExtendedPdfViewerService,
+  PageRenderEvent,
+  AnnotationActionType,
+  ShowCommentTagPopoverDetails,
+  AnnotationDeleteEvent,
+  AnnotationItem,
+  highlightEditor,
+} from 'ngx-extended-pdf-viewer';
+import { CommentPopoverComponent } from './comment-popover/comment-popover.component';
+import { TagPopoverComponent } from './tag-popover/tag-popover.component';
 import {
-  faSliders,
-  faBookOpenReader,
-  faQuoteRight,
-  faArrowTrendUp,
   faTag,
   faTrash,
   faHighlighter,
+  faMessage
 } from '@fortawesome/free-solid-svg-icons';
-import { PdfViewerComponent, PdfViewerModule } from 'ng2-pdf-viewer';
-import { TextOptionsComponent } from './components/text-options/text-options.component';
-import { CommentSelection } from './models/comment-selection.model';
-import { SpanLocation } from './models/span-location.model';
-import { HighlightSelection } from './models/highlight.model';
-import { TagSelection } from './models/tag-selection.model';
-import { HighlightPipe } from './pipes/highlight.pipe';
-import { Nullable } from './types/types';
-import { RIGHT_PANE_NAV } from './models/view.model';
-import { v4 as uuidv4 } from 'uuid';
-import { CommonModule } from '@angular/common';
+import { TagModel, UtilService, TagListModel } from './util.service';
+
 import { FormsModule } from '@angular/forms';
-import { AngularSplitModule } from 'angular-split';
-import { NgbNavModule, NgbTooltipModule } from '@ng-bootstrap/ng-bootstrap';
+import { Subscription } from 'rxjs';
+
 @Component({
   standalone: true,
   imports: [
     CommonModule,
-    PdfViewerModule,
     FontAwesomeModule,
-    FormsModule,
     AngularSplitModule,
     NgbNavModule,
     NgbTooltipModule,
+    NgxExtendedPdfViewerModule,
+    NgbModule,
+    FormsModule,
   ],
+  providers: [provideAnimations(), UtilService],
   selector: 'mo-pdf-viewer',
   templateUrl: './mo-pdf-viewer.component.html',
   styleUrls: ['./mo-pdf-viewer.component.scss'],
 })
-export class MoPdfViewerComponent {
+export class MoPdfViewerComponent implements OnDestroy, OnInit {
   @Input({ required: true }) public pdfSrc: string | Uint8Array = '';
-  @Input() public documentTitle = '';
-  @Input() public documentClassification = 'Unclassified';
-  @Input() public documentAuthor = '';
-  @Input() public minimalView = false;
-  @ViewChild(PdfViewerComponent) public pdfComponent!: PdfViewerComponent;
-  public currentSearchIndex: number = 1;
-  public totalMatchesCount: number = 0;
+  @Input() public set annotations(savedAnnotations: AnnotationItem[]) {
+    if (savedAnnotations && savedAnnotations.length > 0) {
+      this.savedAnnotations = savedAnnotations;
+      this.notRenderedPages = [
+        ...new Set(this.savedAnnotations.map((a) => a.pageNumber)),
+      ];
+
+      this.requestedPages = [...this.notRenderedPages];
+    }
+  }
+  @Input() public savedAnnotations: AnnotationItem[] = [];
+  @Output() public annotationUpdated = new EventEmitter<AnnotationItem[]>();
```

Comment:
Assuming these 2 input/output are linked, they should be named "annotations" and "annotationsChange" to support two way binding.

Reply:
We can't give the name annotations as it is already used here:
`  @Input() public set annotations(savedAnnotations: AnnotationItem[]) {
    if (savedAnnotations && savedAnnotations.length > 0) {
      this.savedAnnotations = savedAnnotations;
      this.notRenderedPages = [
        ...new Set(this.savedAnnotations.map((a) => a.pageNumber)),
      ];

      this.requestedPages = [...this.notRenderedPages];
    }
  }
  @Input() public savedAnnotations: AnnotationItem[] = [];
  @Output() public annotationUpdated = new EventEmitter<AnnotationItem[]>();`

Reply:
The exact name isn't as important as the structure. If an input and an output are for the same value, they should be named `@Input <name> @Output <name>Change` to allow shorthands.